src/intersection-observer.js

```diff
@@ -208,6 +209,10 @@ export class IntersectionObserver extends Observable {
    * @private
    */
   flush_() {
+    if (!this.isElementInDoc_()) {
```

Comment:
It is calling `documentContains(this.baseElement_.win.document, this.baseElement_.element);`. I assume it's not super expensive, but would still need to access win.document, so yes I would prefer not to have it.
But the thing here is `flush_` is delayed by 100ms, by that time element could be removed from document already.
I think what you are suggesting is call `destroy` when that element is removed. So how can I know when and where that's happening,